### DISCUSSION TITLE
Is there a "Creativ Coding" tool in Qualcoder?

### DISCUSSION THREAD

**Discussion by tpotthf** [ccbogel/QualCoder#926] [General]
Hello,

I am working with Qualcoder for the first time and I am generally very impressed with the interface, as it is very minimalist and pragmatic. I also find the AI integration of "kaixxx" excellent.

At the moment, I am analyzing interviews for a grounded theory study, and after the first round of open coding, I now have a multitude of different codes, which I have also annotated with in-text memos. Now, I would like to create categories from the codes I have developed. For example, MaxQDA has the "Creative Coding" function for this purpose. I miss this function or a similar option in Qualcoder. Is there a way to work with the codes in an overview of codes, or would it be possible to implement such a function in the medium term? I primarily work with Linux, and in my opinion, Qualcoder is the best QDA software available for this platform. I would hate to return to MS and MaxQDA.

Sincerely,

Tobi

**Comment by kaixxx** [comment]
Hi Tobi,
can you explain a little more what this "creative coding" function does and what you like about it?

BTW: Did you notice the new version of the AI integration that I've released just a few days ago: https://github.com/kaixxx/QualCoder/tree/ai_integration_rework#readme ? It's a beta version and I'm in the process of testing it. If all goes well, it will be integrated into the official QualCoder soon. 

Best
Kai

**Reply by tpotthf** [reply]
Hey Kai,

I have noticed that after the initial or open coding phase within the framework of working with Grounded Theory, I generate a multitude of codes. The “Creative Coding Tool” within MaxQDA opens a kind of whiteboard where the codes can be dragged and dropped, allowing them to be categorized and related to each other. Subsequently, the category system created there can be applied to the entire project. What I like about this, and what I miss in Qualcoder, is a clear arrangement of the codes to work with them and the associated comments and memos. While it is possible to output these via the report system, it is not possible to work further with them. It seems to me that Qualcoder is more suited for qualitative content analysis, where a category system is already more or less present, whereas in Grounded Theory, this system emerges more from the data itself, making a clear workspace relevant. Personally, I wouldn't necessarily wish for a copy-cat of the “Creative Coding Tool,” but rather a possibility to interact more with codes within Qualcoder. 



About Qualcoder AI:
Thank you for the information regarding the new beta of your Qualcoder fork. I have already tried the previous beta and am curious to see how the development continues. It will certainly be very helpful for my further research. I would have liked to attend your workshop on AI and qualitative research at this year's BMT, but unfortunately, the workshop was already full. If I may, I'll approach you at the BMT and can then explain in more detail what functionality I am missing. 

Sincerly 

Tobi

**Comment by ccbogel** [comment]
Hi @tpotthf 

I am glad you like QualCoder in general and are using Kai's AI functions also.
I have looked at some youtube videos of Maxqda creative coding.
I think I could create something like this where you select the codes, shuffle them around and add in new categories. as well as merge similar codes.

regards Colin

**Reply by kaixxx** [reply]
> I think I could create something like this 

That would be nice! Maybe we could even mix some AI in there, at a later stage...

Regarding the BMT, let me explain: I'm in Berlin right now at an annual meetup of qualitative researchers, the "Berliner Methodentreffen" (BMT). This year, more than 400 people are attending. I'll give a workshop tomorrow about QualCoder and AI and will also be part of a panel discussion about AI in qualitative research, together with Stephan Rädiker from MAXQDA and several other people.  
Apparently, Tobi is also here. It would be nice to talk! You can approach me at any time.

**Comment by ccbogel** [comment]
@tpotthf @kaixxx 
 I have developed a 'creative coding' equivalent function. Under the Coding menu, select Code organiser.
Select a _code branch_ or _All_, then right click to: Add categories, rename codes and categories, update memos, merge codes, merge categories, delete categories.
I have done a lot of testing, but the possibility exists that unexpected errors could still occur. So before using, please make a back up copy of your project, to revert to if needed.
After applying the changed Codes tree, review the coded text, etcetera within the Code text /image/A/V to confirm all appears as it should.
Please let me know if the function works as you anticipated it to be. I could not do a drag link to another code, as is done in MAXQDA, instead I used a selection box window.
regards Colin

**Reply by tpotthf** [reply]
@ccbogel 
Thank you very much, I'll give it a shot in the next couple of days, and then I'll give you my feedback.
